Automattic/wp-calypso
# Generated design picker: Sticky thumbnails

#### Changes proposed in this Pull Request

* As the WordPress logo is inside `SignupHeader` component and it's not easy to stick at the top of the screen, I make it become an `@automatitc/components` and render it inside the `ActionButtons` of the `StepContainer` when we need it to be sticky. We can also update all of the usages to import from `@automatitc/components` in the future.
* Make `ActionButtons` sticky on the desktop
* Make `ActionButtons` sticky with white padding-bottom only when the target positioner is out of the viewport. I also remove the `StickyFooter` to share the same sticky state.
* Make thumbnails to be sticky via CSS
* Make the browser header of the preview toolbar sticky.
* Using`showGeneratedDesigns` for `shouldStickyNavButtons` and `isSticky` for `hasStickyNavButtonsPadding` might be confusing, is there any suggestions for naming 🤔

Note that you can compare the result with https://github.com/Automattic/wp-calypso/pull/64110 which makes the scrolling behavior inside the ifame.

https://user-images.githubusercontent.com/13596067/170662916-9d9d0b3e-4abf-4a46-86c5-489bb4f9cde6.mov

#### Testing instructions



* Go to generated design picker: `/setup/designSetup?siteSlug=<your_site>`
* Scroll down to the navbar, thumbnails, browser header, and continue button are sticky
* Resize and scroll up and down again to check the UI won't be broken
* Switch to another design and scroll up and down to check the UI won't be broken



Related to https://github.com/Automattic/wp-calypso/issues/63794
